src/views/Account/AccountView.vue

```diff
@@ -0,0 +1,171 @@
+        <button class="small-button" @click="onCancelButtonClick">Cancel</button>
+      </li>
+      
+      <li v-if="!isChangingPassword"><strong>Password:</strong> ********</li>
+      <li v-else>
+        <strong>Current Password:</strong> <input type="password" v-model="currentPassword">
+        <strong>New Password:</strong> <input type="password" v-model="newPassword">
+      </li>
+      <li v-if="isChangingPassword">
+        <button class="small-button" @click="onConfirmButtonClick">Confirm</button>
+        <button class="small-button" @click="onCancelButtonClick">Cancel</button>
+      </li>
+    </ul>
+
+    <button @click="onEditProfilePictureButtonClick">Edit Profile Picture</button> 
+    <input type="file" ref="fileInput" @change="onFileInputChange" style="display: none;"> 
+
+    <button @click="onEditProfileInfoButtonClick">Edit Profile Info</button>
+
+    <button @click="onChangePasswordButtonClick">Change Password</button> 
+  </section>
+  <p v-else>Loading...</p>
+</template>
+  
+<script>
+import axios from 'axios';
+import { jwtDecode } from "jwt-decode";
+
+export default {
+  name: 'AccountView',
+  data() {
+    return {
+      user: null,
+      tempUser: null,
+      isEditingProfileInfo: false,
+      isChangingPassword: false,
+      currentPassword: '',
+      newPassword: '',
+    };
+  },
+  async mounted() {
+    const token = localStorage.getItem('jwt');
+    const decodedToken = jwtDecode(token);
+    const userId = decodedToken.data.id;
+
+    try {
+      const response = await axios.get(`${import.meta.env.VITE_API_URL}/users/${userId}`, {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      });
+      this.user = response.data;
+    } catch (error) {
+      console.error(error);
+    }
+  },
+  methods: {
+  onCancelButtonClick() {
+    this.user = this.tempUser;
+    this.currentPassword = '';
+    this.newPassword = '';
+    this.isEditingProfileInfo = false;
+    this.isChangingPassword = false;
+  },
+
+  onChangePasswordButtonClick() {
+    this.tempUser = { ...this.user };
+    this.isChangingPassword = true;
+  },
+
+  async onConfirmButtonClick() {
+    try {
+      const response = await axios.put(`${import.meta.env.VITE_API_URL}/users/changePassword`, {
+        id: this.user.id,
+        currentPassword: this.currentPassword,
+        newPassword: this.newPassword,
+      }, {
+        headers: {
+          Authorization: `Bearer ${localStorage.getItem('jwt')}`,
+        },
+      });
+
+      this.isChangingPassword = false;
+      this.currentPassword = '';
+      this.newPassword = '';
+    } catch (error) {
+      console.error(error);
+    }
+  },
+
+  onEditProfileInfoButtonClick() {
+    this.tempUser = { ...this.user };
+    this.isEditingProfileInfo = true;
+  },
+
+  async onDoneButtonClick() {
+    try {
+      const response = await axios.put(`${import.meta.env.VITE_API_URL}/users/update`, {
```

Comment:
je hoeft de response hier niet bij te houden, je kan gewoon axios.put doen zonder dat allemaal ervoor. als je iets met de response wilt doen is het vaak netter om het zo te doen `axios.put(some url).then((response) = {some code to do with the response})`